pkg/monitoring/poller.go

```diff
@@ -51,10 +54,16 @@ func (s *solanaPollerImpl) Start(ctx context.Context) {
 	}
 
 	for {
-		timer := time.NewTimer(s.fetchInterval)
+		timer := time.NewTimer(s.pollInterval)
 		select {
 		case <-timer.C:
-			data, err := s.reader.Read(ctx, s.account)
+			var data interface{}
+			var err error
+			func() {
```

Comment:
Why this pattern?

Reply:
So I can `defer cancel()` which ensures it will be executed even if Read panics!

Reply:
Defer cancel should always run since you call it before the read. Don't understand the difference of wrapping into a func